components/codec/src/byte.rs

```diff
@@ -1048,12 +1071,11 @@ mod tests {
             let output_len = unsafe {
                 let src_ptr = buffer.as_mut_ptr().add(encoded_prefix_len);
                 let slice_len = buffer.len() - encoded_prefix_len;
-                let src = std::slice::from_raw_parts(src_ptr, slice_len);
-                let dest = std::slice::from_raw_parts_mut(src_ptr, slice_len);
+                let buf = std::slice::from_raw_parts_mut(src_ptr, slice_len);
```

Comment:
I think this is on purpose as L1069 says. There should either be some misunderstanding or the case should be removed.

/cc @breeswish

Reply:
Indeed, how about calling `try_decode_first_internal` instead?